pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/elastic/ElasticSearchService.java

```diff
@@ -74,6 +76,78 @@ public List<String> getAllParticipantGuids(String esParticipantsIndex) {
         return participantIds;
     }
 
+    /**
+     * Get a single participant document based on participant GUID. Throw an exception if document is not found.
+     */
+    public ElasticSearchParticipantDto getRequiredParticipantDocumentById(String ddpParticipantId, String index) {
```

Comment:
Just a suggestion: Do we need both of the words `get` and `required` in the name? If the only difference between this and `getParticipantDocumentById` is the failure when the participant doesn't exist, can we name it something like `getParticipantDocumentByIdOrThrow` or `requireParticipantDocumentById` maybe?

Reply:
Ah, good catch. I added `byId` to disambiguate with another method in this class, but the `byId` methods are now much more widely used. The `byId` adds no value, so I changed the conflicting method name, and got rid of the `byId` part. (The method I wanted to disambiguate is not really an overload of these, so I was reluctant to use the same name for both.)

As for your suggestions, `get` is important and expected here, so I cannot get rid of that. For `orThrow` vs `required`... I've been using `required` for methods like this and phasing out `orThrow`, since the former better describes the intent. That is, I expect the value to be there, so raise and exception if not. The `orThrow`  term is ambiguous (throw for what conditions?), and has a particular and somewhat different meaning when used with `Optional`, so that can be confusing.

Given that, it seems like the term `required` did not resonate with you. Does a better term come to mind?